src/Domain/CoordinateMaps/UniformCylindricalSide.cpp

```diff
@@ -0,0 +1,752 @@
+                                   square(center_one[2] - center_two[2]));
+
+  ASSERT(dist_spheres + radius_one <= 0.98 * radius_two,
+         "The map has been tested only for the case when "
+         "sphere_one is sufficiently contained inside sphere_two, without the "
+         "two spheres almost touching. Radius_one = "
+             << radius_one << ", radius_two = " << radius_two
+             << ", dist_spheres = " << dist_spheres
+             << ", (dist_spheres+radius_one)/radius_two="
+             << (dist_spheres + radius_one) / radius_two);
+
+  const double horizontal_dist_spheres =
+      sqrt(square(center_one[0] - center_two[0]) +
+           square(center_one[1] - center_two[1]));
+
+  if (z_plane_plus_two != z_plane_plus_one and
+      z_plane_minus_two != z_plane_minus_one) {
+    ASSERT(center_one[2] - radius_one <= center_two[2] + 0.2 * radius_two and
+               center_one[2] + radius_one >= center_two[2] - 0.2 * radius_two,
+           "The map has been tested only for the case when "
+           "sphere_one is not a tiny thing at the top or bottom of sphere_two. "
+           " Radius_one = "
+               << radius_one << ", radius_two = " << radius_two
+               << " center_one[2] = " << center_one[2]
+               << " center_two[2] = " << center_two[2]);
+  }
+
+  if (z_plane_plus_two == z_plane_plus_one) {
+    ASSERT(z_plane_minus_two <= z_plane_plus_two - 0.18 * radius_two,
+           "The map has been tested only if the sphere_two planes are far "
+           "enough apart. z_plane_plus_two="
+               << z_plane_plus_two << " z_plane_minus_two=" << z_plane_minus_two
+               << " radius_two=" << radius_two);
+  }
+
+  if (z_plane_minus_two == z_plane_minus_one) {
+    ASSERT(z_plane_plus_two >= z_plane_minus_two + 0.18 * radius_two,
+           "The map has been tested only if the sphere_two planes are far "
+           "enough apart. z_plane_plus_two="
+               << z_plane_plus_two << " z_plane_minus_two=" << z_plane_minus_two
+               << " radius_two=" << radius_two);
+  }
+
+  ASSERT(horizontal_dist_spheres <=
+             (z_plane_plus_two == z_plane_plus_one or
+                      z_plane_minus_two == z_plane_minus_one
+                  ? 0.0
+                  : std::min(radius_one,
+                             std::max(0.0, 0.95 * radius_two - radius_one))),
+         "The map has been tested only for the case when "
+         "sphere_one intersects the polar axis of sphere_two and is not "
+         "too far from the edge of sphere_two (or in the "
+         "case of equal plus or minus z_planes, when the two spheres have the "
+         "same x and y centers)."
+         " Radius_one = "
+             << radius_one << ", radius_two = " << radius_two
+             << ", dist_spheres = " << dist_spheres
+             << ", horizontal_dist_spheres = " << horizontal_dist_spheres);
+
```

Comment:
I am not sure at all.  The numbers in some of the conditions will almost certainly need to be adjusted in the PR that combines the maps into the BBH domain.  I didn't want to go ahead with the domain PR before this one (and the other one #4267 ) were merged, because I don't want to submit a 6000-line PR.